perfmetrics/scripts/load_tests/python/load_generator/load_generator.py

```diff
@@ -0,0 +1,357 @@
+# Copyright 2023 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Load Generator for performing load test with a given task (in python).
+
+Contains the class for generating load with the help of a task written in
+python.
+
+Example:
+
+    task_obj = ReadTask()
+    lg_obj = LoadGenerator(...)
+    lg_obj.pre_load_task(...)
+    observations = lg_obj.generate_load(task_obj)
+    metrics = lg_obj.post_load_task(observations, ...)
+"""
+import logging
+import sys
+import threading
+import time
+import json
+import multiprocessing
+import numpy as np
+from load_generator import constants as lg_const
+
+
+class LoadGenerator:
+  """Generates load with the with a given task.
+
+  Generates load on CPU and other resources depending upon the given task. This
+  class also provides default implementation of post load test task to get 
+  latencies of tasks. Classes derived from this class can define
+  their own pre- and post-load test tasks and run before and after actual load
+  test.
+
+  Args:
+    num_processes: An integer defining the number of processes to run during
+      load test.
+    num_threads_per_process: An integer defining the number of threads to run
+      in each process during load test.
+    run_time: An integer defining the number of seconds to run the load test
+      for.
+    num_tasks_per_thread: An integer defining the number of times the given
```

Comment:
We can rename it to execution_count_per_thread.

Reply:
As discussed, renamed it to num_executions_per_thread